setup.py

```diff
@@ -115,57 +108,67 @@ def finalize_options(self):
         ])
 
 
-setup(
-    name='brainiak',
-    use_scm_version=True,
-    setup_requires=[
-        'cython',
-        # https://github.com/numpy/numpy/issues/14189
-        # https://github.com/brainiak/brainiak/issues/493
-        'numpy!=1.17.*,<1.20',
-        'pybind11>=1.7',
-        'scipy!=1.0.0',
-        'setuptools_scm',
-    ],
-    install_requires=[
-        'cython',
-        # Previous versions fail of the Anaconda package fail on MacOS:
-        # https://travis-ci.org/brainiak/brainiak/jobs/545838666
-        'mpi4py>=3',
-        'nitime',
-        # https://github.com/numpy/numpy/issues/14189
-        # https://github.com/brainiak/brainiak/issues/493
-        'numpy!=1.17.*,<1.20',
-        'scikit-learn[alldeps]>=0.18',
-        # See https://github.com/scipy/scipy/pull/8082
-        'scipy!=1.0.0',
-        'statsmodels',
-        'pymanopt',
-        'theano>=1.0.4',  # See https://github.com/Theano/Theano/pull/6671
-        'pybind11>=1.7',
-        'psutil',
-        'nibabel',
-        'joblib',
-        'wheel',  # See https://github.com/astropy/astropy-helpers/issues/501
-        'pydicom',
-    ],
-    extras_require={
-        'matnormal': [
+extras = {
+    "dev": [
+		"pytest",
+		"coverage",
+		"flake8",
+		"flake8-print",
+		"mypy",
+		"myst-nb",
+		"restructuredtext-lint",
+		"setuptools_scm",
+		"sphinx",
+		"sphinx_rtd_theme",
+		"towncrier",
+		"numdifftools",
+		"testbook"
+	],
+
+	'matnormal': [
             'tensorflow',
-            'tensorflow_probability',
+            'tensorflow_probability<=0.15.0',
         ],
-    },
-    author='Princeton Neuroscience Institute and Intel Corporation',
-    author_email='mihai.capota@intel.com',
-    url='http://brainiak.org',
-    description='Brain Imaging Analysis Kit',
-    license='Apache 2',
-    keywords='neuroscience, algorithm, fMRI, distributed, scalable',
-    long_description=long_description,
+
+	# All requirements for notebook examples in docs/examples
+    "examples": [
+		"nilearn",
+		"nxviz<=0.6.3",
+		"nltools",
+		"timecorr",
+		"seaborn",
+		"holoviews",
+		"pyOpenSSL",
+		"awscli",
+		"bcrypt",
+		"indexed_gzip",
+		"inflect",
+		"ipython",
+		"jupyter",
+                "mypy @ git+https://github.com/python/mypy.git",
```

Comment:
The mypy fix has been released:
https://github.com/python/mypy/issues/13385#issuecomment-1258454516
```suggestion
                "mypy",
```